8.1/startup/generateStartupCommand.js

```diff
@@ -18,15 +18,45 @@ if (typeof process.env.WEBSITE_ROLE_INSTANCE_ID !== 'undefined'
     roleInstanceId = process.env.WEBSITE_ROLE_INSTANCE_ID;
 }
 
-var startupCommand = fs.readFileSync(CMDFILE, 'utf8').trim();
+// Is Application Insights enabled, with an associated ikey?
+var appInsightsEnabled = process.env.ENABLE_APPINSIGHTS && process.env.APPINSIGHTS_INSTRUMENTATIONKEY;
+var appInsightsPreloadArg = "--require /opt/startup/initAppInsights.js";
+var nodeCommandPrefix = "node ";
+
+if (appInsightsEnabled) {
+    console.log("Application Insights enabled");
+    nodeCommandPrefix += appInsightsPreloadArg + " ";
+}
+
+function augmentCommandIfNeccessary(command) {
+    if (!command || !appInsightsEnabled) {
+        return command;
+    }
+
+    // Application Insights is enabled, so we need to
+    // to update the specified startup command to pre-load it.
+    if (command.indexOf("pm2 start ") === 0) {
+        return command += " --node-args='" + appInsightsPreloadArg + "'";
+    } else if (command.indexOf("node ") === 0) {
+        // Simply replacing the prefix allows the user to specify
+        // additional Node flags, in addition to the AI preload one.
+        return command.replace("node ", nodeCommandPrefix);
+    }
+
+    // The command is using an unknown executable, and therefore,
+    // the App Insights runtime can't be automatically enabled.
+    return command;
+}
+
+var startupCommand = augmentCommandIfNeccessary(fs.readFileSync(CMDFILE, 'utf8').trim());
```

Comment:
Besides `pm2 start <process.json>` and `node <startScript>`, are there any other startup command formats that could be provided by the user and passed in when the Docker container is run? I'm not aware of any, but I just wanted to confirm.